bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -310,8 +312,15 @@ public void stateChanged(Item item, State oldState, State newState) {
     @Override
     public void stateUpdated(Item item, State state) {
         State oldState = this.state;
-        if (function != null) {
-            setState(function.calculate(members));
+        if (function != null && baseItem != null) {
+            State calculatedState = function.calculate(members);
+
+            // functions return DecimalTypes so we have to convert them if the baseItem type only supports PercentType

```

Comment:
This seems to be a pretty specific solution... 
Is there anything speaking against using `ItemUtil.convertToAcceptedState(State, Item)` here?